File: bindings/junit/ExampleTest.java
package com.magnetoserge.junit;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import static com.magnetoserge.junit.MagnetoAssertions.*;

/**
 * Example tests using Magnéto-Serge JUnit assertions
 */
public class ExampleTest {

    private static HttpClient client;

    @BeforeAll
    public static void setUp() {
        // Configure HTTP client to use Magnéto proxy
        client = HttpClient.newBuilder()
                .proxy(java.net.ProxySelector.of(
                        new java.net.InetSocketAddress("localhost", 8888)
                ))
                .build();

        // Set cassette directory
        MagnetoAssertions.setCassetteDirectory("./cassettes");
    }

    @Test
    public void testUserLogin() throws Exception {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("http://localhost:8080/api/authenticate"))
                .POST(HttpRequest.BodyPublishers.ofString("{\"username\":\"admin\",\"password\":\"admin\"}"))
                .header("Content-Type", "application/json")
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        // Assert response matches cassette
        assertMatchesCassette(response, "user-login");
    }

    @Test
    public void testUserAccount() throws Exception {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("http://localhost:8080/api/account"))
                .GET()
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        // Assert status matches cassette
        assertCassetteStatus(response, "user-account", 200);
    }

    @Test
    public void testInteractionCount() {
        // Assert cassette has expected number of interactions
        assertInteractionCount("user-login", 3);
    }

    @Test
    public void testCassetteCookies() {
        // Assert cassette contains cookies
        assertHasCookies("user-login");
    }

    @Test
    public void testSessionCookie() {
        // Assert cassette has specific cookie
        assertHasCookie("user-login", "JSESSIONID");
    }

    @Test
    public void testCassetteVersion() {
        // Assert cassette version
        assertCassetteVersion("user-login", "2.0");
    }
}

File: bindings/phpunit/ExampleTest.php
<?php

namespace MagnetoSerge\PHPUnit\Tests;

use MagnetoSerge\PHPUnit\MagnetoAssertions;
use PHPUnit\Framework\TestCase;
use GuzzleHttp\Client;

/**
 * Example tests using Magnéto-Serge PHPUnit assertions
 */
class ExampleTest extends TestCase
{
    use MagnetoAssertions;

    private Client $client;

    protected function setUp(): void
    {
        parent::setUp();

        // Configure HTTP client to use Magnéto proxy
        $this->client = new Client([
            'proxy' => 'http://localhost:8888',
            'base_uri' => 'http://localhost:8080',
        ]);

        // Set cassette directory
        self::setCassetteDirectory('./cassettes');
    }

    public function testUserLogin(): void
    {
        $response = $this->client->post('/api/authenticate', [
            'json' => [
                'username' => 'admin',
                'password' => 'admin',
            ],
        ]);

        // Assert response matches cassette
        $this->assertMatchesCassette($response, 'user-login');
    }

    public function testUserAccount(): void
    {
        $response = $this->client->get('/api/account');

        // Assert status matches cassette
        $this->assertCassetteStatus($response, 'user-account', 200);
    }

    public function testInteractionCount(): void
    {
        // Assert cassette has expected number of interactions
        $this->assertInteractionCount('user-login', 3);
    }

    public function testCassetteCookies(): void
    {
        // Assert cassette contains cookies
        $this->assertHasCookies('user-login');
    }

    public function testSessionCookie(): void
    {
        // Assert cassette has specific cookie
        $this->assertHasCookie('user-login', 'JSESSIONID');
    }

    public function testCassetteVersion(): void
    {
        // Assert cassette version
        $this->assertCassetteVersion('user-login', '2.0');
    }
}
